### PR TITLE
PSD-1168 Search filters case type

### PR DESCRIPTION
**Pull request by samlionheart**
Case type filters, checkboxes for allegation, enquiry and project.
## Description

Checkboxes for case types. When no checkboxes are selected it shows all cases, when all checkboxes are selected it shows all cases.
## Checklist:

- [x] Automated checks are passing locally.
- [x] CHANGELOG updated if change is worth telling users about.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,10 +3,7 @@ All notable changes to this project will be documented in this file.
 
 ## Unreleased
 <!-- ### Product safety database -->
-- New service navigation
-- Welcome email
-- Product search
-- Cookie banner
```
Comment (tom-bell-softwire) [inline CHANGELOG.md]: I don't think these changes have been released yet, so worth keeping them here.

File: psd-web/app/helpers/investigations_helper.rb [hunk 2]
```diff
@@ -34,6 +35,17 @@ def get_status_filter
     { must: { term: status } }
   end
 
+  def get_type_filter
+    return {} if params[:allegation] == "unchecked" && params[:enquiry] == "unchecked" && params[:project] == "unchecked"
+
+    array_of_types = []
```
Comment (tom-bell-softwire) [inline psd-web/app/helpers/investigations_helper.rb]: You could probably just call this `types` (the `array_of_` is a bit redundant).

**Comment by samlionheart** [comment]
Thanks for the comments Tom.

Couple of things I missed:

The filters for assignee and case type were "should", they actually need to be "must". I've changed them all to "must" as they should be AND filters. Because they are all "must", when merged they need to add together instead of overwrite each other.

**Comment by tom-bell-softwire** [comment]
Thanks @samlionheart ! How does the AND filter requirement work within a given filter – i.e., if I select both "allegation" and "enquiry", that should be an OR query, right?

**Pull request closed without merging**